javascript-bank/account.js

```diff
@@ -0,0 +1,41 @@
+/* exported Account */
+function Account(number, holder) {
+  this.number = number;
+  this.holder = holder;
+  this.transactions = [];
+}
+Account.prototype.deposit = function (amount) {
+  var deposit = new Transaction('deposit', amount);
+  if (amount && typeof amount !== 'string' && amount > 0 && Number.isInteger(amount)) {
+    this.transactions.push(deposit);
+    return true;
+  } else {
+    return false;
+  }
+};
+
+Account.prototype.withdraw = function (amount) {
+  var withdraw = new Transaction('withdrawal', amount);
+  if (amount && typeof amount !== 'string' && amount > 0 && Number.isInteger(amount)) {
```

Comment:
You can remove the first two expressions in this condition, you just need to check if it is greater than `0` and an integer, checking if it is an integer will account for if it is a `number` data type. Also, if you want to check if a value is a `numb er`, you should check if its data type **_is `number`_**, not "not a string" 😬